components/infra-proxy-service/cmd/infra-proxy-service/commands/serve.go

```diff
@@ -0,0 +1,115 @@
+package commands
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/chef/automate/components/infra-proxy-service/server"
+	"github.com/chef/automate/components/infra-proxy-service/service"
+	"github.com/chef/automate/components/infra-proxy-service/storage/postgres/migration"
+	"github.com/chef/automate/lib/grpc/secureconn"
+	"github.com/chef/automate/lib/logger"
+	platform_config "github.com/chef/automate/lib/platform/config"
+	"github.com/chef/automate/lib/tls/certs"
+	"github.com/chef/automate/lib/tracing"
+)
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Launches the automate infra proxy service.",
+	Run:   serve,
+	Args:  cobra.ExactArgs(1),
+}
+
+type config struct {
+	GRPC            string `mapstructure:"grpc"`
+	LogFormat       string `mapstructure:"log-format"`
+	LogLevel        string `mapstructure:"log-level"`
+	certs.TLSConfig `mapstructure:"tls"`
+	PGURL           string `mapstructure:"pg_url"`
+	Database        string `mapstructure:"database"`
+	MigrationsPath  string `mapstructure:"migrations-path"`
+}
+
+func serve(cmd *cobra.Command, args []string) {
+	cmd.PersistentFlags().StringP("log-level", "l", "info", "log level")
+	cmd.PersistentFlags().StringP("log-format", "f", "text", "log format")
+	cmd.PersistentFlags().String("grpc", "127.0.0.1:10153", "grpc host and port")
+	cmd.PersistentFlags().StringP("pg_url", "p", "", "postgres uri")
+	cmd.PersistentFlags().StringP("migrations-path", "m", "", "migrations path")
+
+	viper.SetConfigFile(args[0])
+	if err := viper.ReadInConfig(); err != nil {
+		fail(errors.Wrap(err, `Could not read config file. Please pass a config file as the only argument to this command.`))
+	}
+
+	cfg := config{}
+	if err := viper.Unmarshal(&cfg); err != nil {
+		fail(errors.Wrap(err, "couldn't parse configuration file"))
+	}
+
+	if cfg.PGURL == "" {
+		var err error
+		cfg.PGURL, err = platform_config.PGURIFromEnvironment(cfg.Database)
+		if err != nil {
+			fail(errors.Wrap(err, "Failed to get pg uri"))
+		}
+	}
+
+	closer, err := tracing.NewGlobalTracer("infra-proxy")
```

Comment:
we dont use any of this tracing stuff. We should probably stop putting it in

Reply:
Okay sure, will check and update it accordingly thanks pointing it out.